src/foreign/API.hs

```diff
@@ -70,12 +69,11 @@ dexEval ctxPtr sourcePtr = do
 
 dexLookup :: Ptr Context -> CString -> IO (Ptr (WithErr APIErr (Ptr Atom)))
 dexLookup ctxPtr namePtr = do
-  Context (EvalConfig _ backend _) env <- deRefStablePtr $ castPtrToStablePtr $ castPtr ctxPtr
+  Context opts env <- deRefStablePtr $ castPtrToStablePtr $ castPtr ctxPtr
   name <- peekCString namePtr
   case envLookup env (GlobalName $ fromString name) of
     Just (_, LetBound _ (Atom atom)) -> do
-      val <- substArrayLiterals backend atom
```

Comment:
Yes, they're closed and we do have pointer literals. We just abstract them as arguments to Imp/LLVM programs here: https://github.com/google-research/dex-lang/blob/list-code-gen/src/lib/Imp.hs#L80 so that they don't appear in Imp/LLVM programs.